File: app/src/main/java/com/dhl/wanandroid/activity/KnowledgeInfoActivity.java
package com.dhl.wanandroid.activity;

import android.app.Activity;
import android.content.Intent;
import androidx.annotation.Nullable;
import com.google.android.material.tabs.TabLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.appcompat.widget.Toolbar;
import android.view.MenuItem;

import com.dhl.wanandroid.R;
import com.dhl.wanandroid.fragment.KnowledgeTabFragment;
import com.dhl.wanandroid.model.KnowledgeInfo;
import com.dhl.wanandroid.model.KnowledgeInfochild;

import java.util.ArrayList;
import java.util.List;


/**
 * 知识体系 详情
 */
public class KnowledgeInfoActivity extends AppCompatActivity {


    private Toolbar toolbar ;
    private TabLayout tabLayout ;

    private ViewPager viewPager ;

    /**
     * TabLayout title
     */
    private List<String> indicators ;

    private List<KnowledgeTabFragment> fragmentList ;

    private KnowledgeInfo knowledgeInfo ;

    public static  void startActivity(Activity activity , KnowledgeInfo knowledgeInfo)
    {
        Intent intent = new Intent(activity,KnowledgeInfoActivity.class);
        intent.putExtra("knowledgeInfo",knowledgeInfo);
        activity.startActivity(intent);
    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_knowledge_info);
        initView();
        initData();
    }

    private void initView()
    {
        toolbar = findViewById(R.id.tool_bar);
        //toolbar.setNavigationIcon(R.mipmap.toolbar_back);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);
        tabLayout = findViewById(R.id.tab_layout);
        viewPager = findViewById(R.id.content_vp);
    }
    private void initData()
    {
        indicators = new ArrayList<>();
        fragmentList = new ArrayList<>();
        Intent intent = getIntent();
        if(intent != null)
        {
            knowledgeInfo = (KnowledgeInfo) intent.getSerializableExtra("knowledgeInfo");
            toolbar.setTitle(knowledgeInfo.getName());
            for(KnowledgeInfochild knowledgeInfochild :knowledgeInfo.getChildren())
            {
                indicators.add(knowledgeInfochild.getName());
                fragmentList.add(KnowledgeTabFragment.newInstance(knowledgeInfochild.getName(),knowledgeInfochild.getCid()+""));
            }
            viewPager.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager()) {
                @Override
                public Fragment getItem(int i) {
                    return fragmentList.get(i);
                }

                @Override
                public int getCount() {
                    return fragmentList.size();
                }

                @Nullable
                @Override
                public CharSequence getPageTitle(int position) {
                    return indicators.get(position);
                }
            });
            tabLayout.setupWithViewPager(viewPager);
            viewPager.setOffscreenPageLimit(fragmentList.size());
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId())
        {
            case android.R.id.home:
                finish();
                break;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/dhl/wanandroid/activity/KnowledgeInfoActivity.kt
package com.dhl.wanandroid.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.dhl.wanandroid.R
import com.dhl.wanandroid.fragment.BaseFragment
import com.dhl.wanandroid.fragment.KnowledgeTabFragment
import com.dhl.wanandroid.model.KnowledgeTreeData
import com.dhl.wanandroid.util.SettingUtil
import com.google.android.material.tabs.TabLayout

/**
 * 知识体系 详情
 * @author dhl
 * @date 2022 12-28
 */
class KnowledgeInfoActivity : BasicActivity() {

    private val tabLayout: TabLayout by lazy {
        findViewById(R.id.tab_layout)
    }

    private val viewPager: ViewPager by lazy {
        findViewById(R.id.content_vp)
    }

    /**
     * TabLayout title
     */
    private var indicators: MutableList<String> = mutableListOf()

    private var fragmentList: MutableList<KnowledgeTabFragment> = mutableListOf()

    private lateinit var knowledgeTreeData: KnowledgeTreeData

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_knowledge_info)
        initView()
        initData()
    }

    private fun initView() {
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)
    }

    override fun onResume() {
        super.onResume()
        tabLayout.setBackgroundColor(SettingUtil.getColor())
    }

    /**
     * 初始化数据
     */
    private fun initData() {
        intent.let {
            knowledgeTreeData = it.getSerializableExtra("KnowledgeTreeData") as KnowledgeTreeData
            supportActionBar?.title = knowledgeTreeData.name
            for ((_, _, _, _, id, _, _, name) in knowledgeTreeData.children) {
                indicators.add(name)
                fragmentList.add(BaseFragment.newInstance(name, id.toString() + ""))
            }
            viewPager!!.adapter = object : FragmentPagerAdapter(supportFragmentManager) {
                override fun getItem(i: Int): Fragment {
                    return fragmentList[i]
                }

                override fun getCount(): Int {
                    return fragmentList.size
                }

                override fun getPageTitle(position: Int): CharSequence? {
                    return indicators[position]
                }
            }
            tabLayout.setupWithViewPager(viewPager)
            viewPager.offscreenPageLimit = fragmentList.size
        }

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> finish()
        }
        return super.onOptionsItemSelected(item)
    }

    /**
     * 启动传参
     */
    companion object {
        fun startActivity(activity: Activity, knowledgeTreeData: KnowledgeTreeData?) {
            val intent = Intent(activity, KnowledgeInfoActivity::class.java)
            intent.putExtra("KnowledgeTreeData", knowledgeTreeData)
            activity.startActivity(intent)
        }
    }
}